ext/standard/tests/serialize/unserialize_extra_data.phpt

```diff
@@ -0,0 +1,26 @@
+--TEST--
+Extra data at the end of a valid value
+--FILE--
+<?php
+
+var_dump(unserialize('i:5;i:6;'));
+var_dump(unserialize('N;i:6;'));
+var_dump(unserialize('b:1;i:6;'));
+var_dump(unserialize('a:1:{s:3:"foo";b:1;}i:6;'));
```

Comment:
I suppose it will also warn for nested data in an array?
``var_dump(unserialize('a:1:{s:3:"foo";b:1;s:3:"bar";b:1;}'));``
(I think, first time staring at PHP's unserializing format).

Reply:
That is already a hard syntax error: https://3v4l.org/e15dB.

What this PR does is: If there is a successful parse of a value (whatever it is) and there are unconsumed left-over bytes in the input string, then a warning is emitted. In other words: Whenever data is appended to correctly serialized data.

Reply:
Huh TIL. In any case this seems sensible to me. :)